src/lava/magma/core/model/py/model.py

```diff
@@ -419,7 +419,8 @@ def add_ports_for_polling(self):
                     if isinstance(csp_port, CspRecvPort):
                         def func(fvar_port=var_port):
                             return lambda: fvar_port
-                        self._channel_actions.append((csp_port, func(var_port)))
+                        self._channel_actions.insert(0,
```

Comment:
So rather than appending to the `_channel_actions` list 
```
self._channel_actions: ty.List[ty.Tuple[ty.Union[CspSendPort, CspRecvPort], ty.Callable]]
```
we are overwriting idx 0.

Reply:
We are prepending an element rather than appending.

Reply:
Yes, insert is not overwriting existing elements.

Reply:
My mistake. 👍🏾 

Appending.